### ISSUE TITLE
Sprite drawing bug (upstream bug)

### ISSUE THREAD

**Issue by Tatsuya79** [libretro/px68k-libretro#24]
Not a libretro bug, it's present in both winx68k and px68k upstream.
Referencing it here as there's no development going on upstream.

I saw it in several games but the one I can remember is Gardis Light 2.10 (not present in older versions), the one where you can choose between 3 characters.

The sprites aren't redrawn properly. 
Try jumping and you'll see the character staying in the air sometimes, while he should normally be on the ground. 
Same for swinging your sword: the last part of the sprite animation with sword in your hand stays on if you don't move.
It seems to happen when several sprites move at the same time.

XM6 Type G doesn't have this issue.

**Comment by retro-wertz** [comment]
i cant seem to replicate this, though i havent really played past 3-4 change screens(why are px68k games seems hard?) is there specific stage, timing this occurs?

**Comment by Tatsuya79** [comment]
It happens everywhere. You can stay on the 1st screen and just push the jump button sometimes without moving. Compare with previous version then if you don't understand what the issue is.

It's an easy game, I finished it twice already! But haven't managed to save my partner yet.

**Comment by negativeExponent** [comment]
was there a variant of Gardis without this sprite bug? or does all of them have it.

**Comment by Tatsuya79** [comment]
I have 2 versions without the issue: "2.02" and "2.02 alt".

Then the version "2.10" with a 3rd character is the one with redraw bugs.

**Comment by Tatsuya79** [comment]
I found another spot with this kind of sprite drawing issue.

In Etoile Princesse, 4th stage the fire mountain, the dragon boss at the end has his upper body disappearing, following the movement of its head which is another sprite.

![Etoile Princesse-240915-213819](https://github.com/user-attachments/assets/fbb52535-d7c7-4c1d-ad25-0a19b9bff8ae)
![Etoile Princesse-240915-213824](https://github.com/user-attachments/assets/9c2a7f2f-b544-456d-86ab-08400500764a)

**Comment by negativeExponent** [comment]
can you post the savestate for this area? or probably the savefile/the romfile for the savedata?

**Comment by Tatsuya79** [comment]
Savestate for px68k from this repo, game is the hdf version.

[Etoile Princesse.zip](https://github.com/user-attachments/files/17007775/Etoile.Princesse.zip)

**Comment by negativeExponent** [comment]
huh. the game is just reboots when loading the state. our games probably not compatible

**Comment by Tatsuya79** [comment]
[ETOILE PRINCESSE.zip](https://github.com/user-attachments/files/17010654/ETOILE.PRINCESSE.zip)

If the savestate isn't working, the 2nd option in the menu is for loading the game at that stage.

https://youtu.be/ejUN15fYgYs?t=1857

**Comment by negativeExponent** [comment]
nope. nothing appears. i was expecting that to happen. but no save data is loaded

**Comment by Tatsuya79** [comment]
I don't get it, you're not confusing px68k vs px68k_next paths or something like that?
Also I shared all that with different caps that can cause issue in the names.

I tested again on the latest core from here an all is fine. I'm re-uploading everything here with better naming:
[Etoile.zip](https://github.com/user-attachments/files/17011596/Etoile.zip)

**Comment by negativeExponent** [comment]
this works now. still stateloading reboots the game. win10/linux state incompatible?

**Comment by Tatsuya79** [comment]
It loads fine but after 2s it get stuck like that:

![Etoile Princesse-240916-125838](https://github.com/user-attachments/assets/85ac2ceb-3533-4317-8acc-919a1b759eb8)

**Comment by negativeExponent** [comment]
yeah... your state from windows works only in windows, my state from linux works only in linux. tested on Wine. has there been cross-platform incompatibility with states before? im using state module from beetle, was hoping both bigendian (if we can manage to have it tested) and cross-platform will not be an issue.

**Comment by Tatsuya79** [comment]
Suggestions I can get:

are they compiled with all of the same settings?

**Comment by LibretroAdmin** [comment]
I just asked themaister. There is no cross-platform portability guaranteed with savestates. It's entirely up to the core to guarantee they are so.

**Comment by negativeExponent** [comment]
not a clue... as far as platform-specific code. issue probably are some of the callbacks, like IRQ refers to function addresses.

**Comment by negativeExponent** [comment]
wine appears to segfault when loading a state from linux. it shows some thing like this:

https://gist.githubusercontent.com/negativeExponent/c46caa70862ff030fed6b6ae307d28e5/raw/49094c84b0b10fc14c16a0e1ae507976f86dbbcb/gistfile1.txt

not sure how to understand that though. (im using my fork here, which has the same issue)

also adress error occurs in px68k when you wait awhile.

**Comment by negativeExponent** [comment]
> Xer Shadow Tail 🌺 — Today at 12:59 `It is probably long`

dunno if this was intended....

anyways can you test this on the same hdf file? state2

[Etoile Princesse.state2.zip](https://github.com/user-attachments/files/17028556/Etoile.Princesse.state2.zip)
